### ISSUE TITLE
plugin/bind stopped working (listen on non assigned IPs on loopback device)

### ISSUE THREAD

**Issue by gebi** [miekg/coredns#5133]
**What happened**:
After upgrade to 1.8.7 plugin/bind stopped working

```
coredns[58398]: plugin/bind: failed to get interfaces list: route ip+net: netlinkrib: address family not supported by protocol
```

**What you expected to happen**:

Working, like in eg. 1.6.3

```
Jan 30 23:45:13 x coredns[61752]: .:53 on 127.0.1.53
...
Jan 30 23:45:13 x coredns[61752]:   / ____/___  ________  / __ \/ | / / ___/        ~ CoreDNS-1.6.3
```

**How to reproduce it (as minimally and precisely as possible)**:

Our config (to not create clashes when different DNS caches are required on the machine, sadly yes)
```
. {
    bind 127.0.1.53
    errors
    forward . FWD_IP1 FWD_IP2 FWD_IPx {
        force_tcp
    }
    cache 600
    prometheus 127.0.0.1:9953
}
```

**Anything else we need to know?**:

culprint maybe #4219 

**Environment**:

- debian buster
- upstream coredns release packaged as `.deb`
- the version of CoreDNS: 1.8.7
- Corefile: s.a
- logs, if applicable: s.a

**Comment by chrisohaver** [comment]
Error message was merged in #4543.

Looks like #4543 adds a new option - but (unnecessarily?) does a bunch of additional setup leg work to support that option when the new option is not used.  So it seems that at least this bug could made smaller in scope by only doing those things if the option that requires them is actually used in the Corefile.  Better would be to fix the option too, but at least it shouldn't error when the option is not in use.

**Comment by chrisohaver** [comment]
Looking at this more closely, it would have been introduced in #4522, not #4543.
